.github/workflows/pr.yaml

```diff
@@ -170,8 +170,8 @@ jobs:
     with:
       build_type: pull-request
       script: "ci/test_python_scikit_learn_tests.sh"
-      # Select the amd64 entry with the highest CUDA and Python version
-      matrix_filter: map(select(.ARCH=="amd64")) | [max_by([.CUDA_VER,.PY_VER]|map(split(".")|map(tonumber)))]
+      # One run for each dependencies config on amd64, breaking ties by highest Python version
+      matrix_filter: map(select(.ARCH == "amd64")) | sort_by(.PY_VER) | unique_by(.DEPENDENCIES)
```

Comment:
Used play.jqlang.org to fiddle this out, I think it does what I want (one consistent run each latest and oldest deps): https://play.jqlang.org/s/HdpY_YU6EnledXL